src/notification/notification.service.ts

```diff
@@ -0,0 +1,66 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import * as firestoreAdmin from 'firebase-admin/firestore';
+import { messaging } from 'firebase-admin';
+import { Messaging } from 'firebase-admin/lib/messaging/messaging';
+import { Device, NotificationMessage } from './interface/notification.interface';
+import { Message, MulticastMessage } from 'firebase-admin/lib/messaging/messaging-api';
+@Injectable()
+export class NotificationService {
+  private collection: firestoreAdmin.CollectionReference;
+  private messenger: Messaging;
+  constructor() {
+    this.collection = firestoreAdmin.getFirestore().collection('aegis-notification');
+    this.messenger = messaging();
+  }
+  async validateRegistrationToken(registrationToken: string): Promise<boolean> {
+    try {
+      return !!(await this.messenger.send({ token: registrationToken }, true));
+    } catch (err) {
+      return false;
+    }
+  }
+  async registerDevice(uid: number, registrationToken: string): Promise<boolean> {
+    if (!this.validateRegistrationToken(registrationToken)) throw new BadRequestException('Invalid token');
+    const deviceRef = this.collection.doc(uid.toString());
+    const res = await deviceRef.set({
+      token: registrationToken,
+      timestamp: Date.now(),
+    });
+    return !!res;
```

Comment:
this too

Reply:
To cast the OK response into boolean. Actually i'll move this to controller level.